### DISCUSSION TITLE
tfjs-node module exceed max serverless function size, 312 MB uncompressed / 74.1 MB compressed size

### DISCUSSION THREAD

**Discussion by helloivanco** [opensource9ja/danfojs#256] [Q&A]
Using danfojs-node in a NextJS project which I'm trying to deploy via Vercel.  

I'm importing dfd via `const dfd = require('danfojs-node');` 

Unable to deploy build as the node_modules/@tensorflow/tfjs-node exceeds the max serverless function size of 50 MB compressed or 250 MB uncompressed. Everything works fine in dev.

Is there a way to reduce the module size / workaround to this issue? 

<img width="965" alt="Screen Shot 2021-07-26 at 15 55 54 PM" src="https://user-images.githubusercontent.com/70876305/127012860-a5d4ccf9-1418-4415-ae11-89103577bd58.png">

Thank you.

**Comment by risenW** [comment]
@hellopaidco Sadly we can't do anything about that size, since it's an external package. Just a thought, since you using Nextjs, why not use danfojs browser version? It's much smaller. https://www.npmjs.com/package/danfojs

**Reply by helloivanco** [reply]
@risenW - I'll keep the browser version in mind. Thank you.

**Reply by risenW** [reply]
@hellopaidco Did you try named import instead of importing the whole lib. For example you could import just functions or classes you need.

This helps in tree shaking and can reduce size.

**Reply by Gnpd** [reply]
Hello @risenW , in my case I am using the client version in Next.js. The build is as follows:

no danfojs  -->  6.52 kB 
using `import * as dfd from "danfojs"` --> 1.97 MB 
using `import { Series } from 'danfojs' `--> 1.97 MB
using `import { Series } from 'danfojs/dist/danfojs-base' `--> 1.62 MB
